internal/route/request_executor.go

```diff
@@ -34,18 +35,31 @@ func (e *HandledError) Error() string {
 }
 
 type OriginError struct {
-	err error
+	err          error
+	response     string
+	ResponseCode int
+}
+
+type HttpResponse struct {
+	StatusCode int
 }
 
 func (e *OriginError) Error() string {
 	return fmt.Sprintf("error making request to origin: %v", e.err)
 }
 
+/*
+ * Return arguments
+ * 1. JSON Response body - Body of response decoded into JSON RPC, if possible.
+ * 2. HTTP Response - Raw HTTP response from upstream, if possible.
+ * 3. Cached - Whether the response was cached.
+ * 4. Error - Error encountered when making request to upstream or another error encountered during processing.
+ */
 func (r *RequestExecutor) routeToConfig(
 	ctx context.Context,
 	requestBody jsonrpc.RequestBody,
 	configToRoute *config.UpstreamConfig,
-) (jsonrpc.ResponseBody, *http.Response, bool, error) {
+) (jsonrpc.ResponseBody, *HttpResponse, bool, error) {
```

Comment:
Didn't think it was necessary to plumb the full `*http.Response` object all the way back to the front.  We can `close` it and pass back some specific pieces of metadata we're interested in (e.g. httpStatusCode).

The impetus for this change was that I was getting confused when I was first looking at the code (I haven't looked at it in months!).